code/datums/datum.dm

```diff
@@ -278,3 +282,72 @@
 /datum/proc/GenerateTag()
 	datum_flags |= DF_USE_TAG
 
+/datum/proc/add_filter(name,priority,list/params)
```

Comment:
i dont understand, why are we doing this?

Reply:
oh for images. ugh

Reply:
this is extremely evil code and most likely not necessary at all to get this pr working

Reply:
it's a side effect of oop only built on inheritance chains being terrible. most i could ask for on this is `TEST_ONLY_ASSERT(isatom(src) || istype(src, /image))` I think, which would at least highlight why this is being done beter